src/towncrier/_builder.py

```diff
@@ -349,6 +349,17 @@ def render_fragments(
     data: dict[str, dict[str, dict[str, list[str]]]] = {}
     issues_by_category: dict[str, dict[str, list[str]]] = {}
 
+    def maybe_append_newlines(text: str) -> str:
+        # if a newsfragment text ends with a code block, we want to append two newlines
```

Comment:
I think is best to have this comment as a docstring and explain why use case for this function.

and maybe move it outside of `render_fragments`.

The TODOs needs to be fixed before merge.
If they can't be fixed, we should have followup GitHub Issues created for each TODO and the comments should contain a link to the GitHub Issue

---------

I know that we have `get_indent` that is embedded and has a comment instead of docstring... but I think (subjective opinion) that is a bad code practice

Reply:
Moved it outside of `render_fragments`, added docstring, and added a bunch of comments explaining the regex logic. Which is also very much changed because I realized it was horribly broken >.>

Personally I don't think embedded functions are a bad practice if the function is only intended as a helper for that one specific function and is exceedingly unlikely to be used by any other function, but having it outside is plenty fine. Mentioned it only being used by `render_fragments` in its docstring.